Lib/glyphsLib/builder/glyph.py

```diff
@@ -35,9 +35,16 @@ def to_ufo_glyph(self, ufo_glyph, layer, glyph_data):
     if last_change is not None:
         ufo_glyph.lib[GLYPHLIB_PREFIX + 'lastChange'] = to_ufo_time(last_change)
     color_index = glyph_data.color
-    if color_index is not None and color_index >= 0:
+    if color_index is not None:
         ufo_glyph.lib[GLYPHLIB_PREFIX + 'ColorIndex'] = color_index
-        ufo_glyph.lib[PUBLIC_PREFIX + 'markColor'] = GLYPHS_COLORS[color_index]
+        color_tuple = None
+        if isinstance(color_index, list):
+            color_tuple = ','.join(['{0:.4f}'.format(i/255.0)
+                                    if i in range(1, 255) else str(int(i/255)) for i in color_index])
+        elif isinstance(color_index, int) and color_index in range(len(GLYPHS_COLORS)):
+            color_tuple = GLYPHS_COLORS[color_index]
+        if color_tuple is not None:
```

Comment:
Shouldn’t we at least log a warning that we dropped some data even though invalid or unsupported that we could not fit?

Reply:
Well, the parser won't allow most invalid things to even get to this point, but it is possible to manually put in some bad values like a tuple of (344, -400, 23, 1) so we can warn for that.  A Glyphs color index of > 11 will fail because GLYPHS_COLORS only has 12 elements. I put in some warnings for either of those cases.